csms.py

```
@@ -0,0 +1,712 @@
+  os.replace(tmp, path)
+
+
+def post_json_cached(url, payload, ttl=pipeline.CACHE_TTL):
+  """POST a JSON body and cache the response on disk, keyed by URL + payload.
+
+  pipeline.github_request_cached is GET-only and picks its cache extension from
+  the Accept header, so the OCA endpoint needs its own helper. Shares the same
+  cache directory, which is already git-ignored.
+  """
+  # The payload is part of the key: one URL serves every page and filter.
+  key = hashlib.md5((url + json.dumps(payload, sort_keys=True)).encode()).hexdigest()
```

Ruff (0.16.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)